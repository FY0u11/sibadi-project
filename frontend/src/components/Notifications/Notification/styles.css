.my-notification {
  margin-top: 15px;
  width: 350px;
  opacity: 0.9;
  border: 0;
  box-shadow: 0px 0px 10px rgba(100, 100, 100, 0.9);
}
