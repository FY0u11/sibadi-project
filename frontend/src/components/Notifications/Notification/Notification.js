import React from 'react'

import './styles.css'

const colors = {
  dark: {
    bg: 'bg-dark',
    color: 'text-light'
  },
  warning: {
    bg: 'bg-warning',
    color: 'text-dark'
  },
  danger: {
    bg: 'bg-danger',
    color: 'text-light'
  },
  success: {
    bg: 'bg-success',
    color: 'text-light'
  }
}

const Notification = ({ level, title, message, closeNotification }) => {
  return (
    <div className="card my-notification">
      <h5 className={`card-header ${colors[level].bg} ${colors[level].color}`}>
        {title}
        <button type="button" className="ml-2 mb-1 close" onClick={closeNotification}>
          &times;
        </button>
      </h5>
      <div className="card-body">
        <p className="card-text">{message}</p>
        <button className={`btn btn-${level} ${colors[level].color}`} onClick={closeNotification}>
          Close
        </button>
      </div>
    </div>
  )
}

export default Notification
