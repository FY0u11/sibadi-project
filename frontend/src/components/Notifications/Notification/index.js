import React, { useEffect } from 'react'

import timer from '../../../utils/timer'

import Notification from './Notification'

// delay before notifications are automatically closed (in milliseconds)
const NotificationContainer = ({ id, delay = 10000, closeNotification, ...rest }) => {
  useEffect(() => {
    timer(delay)(closeNotification, id)
  }, [])

  return <Notification {...rest} closeNotification={() => closeNotification(id)} />
}

export default NotificationContainer
