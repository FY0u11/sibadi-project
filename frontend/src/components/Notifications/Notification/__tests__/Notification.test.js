import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Notification from '../Notification'
import NotificationContainer from '../'

afterEach(cleanup)

describe('Testing Notification component', () => {
  it('It should be properly rendered', () => {
    const { asFragment } = render(
      <Notification title={'Test notification'} level="success" message="Test message" />
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('It should trigger callback two times', () => {
    const callback = jest.fn()
    const { getByText } = render(<Notification level={'success'} closeNotification={callback} />)
    const button1 = getByText('Ã—')
    const button2 = getByText('Close')
    fireEvent.click(button1)
    fireEvent.click(button2)
    expect(callback).toBeCalledTimes(2)
  })
  it('The callback should be automatically triggered in 200 ms', async () => {
    const callback = jest.fn()
    const { getByText } = render(
      <NotificationContainer
        id={'Some id'}
        delay={0.2}
        level={'success'}
        closeNotification={callback}
      />
    )
    expect(callback).toBeCalledTimes(0)
    await new Promise(resolve => setTimeout(resolve, 200))
    fireEvent.click(getByText('Close'))
    expect(callback).toBeCalledTimes(2)
    expect(callback).toBeCalledWith('Some id')
  })
})
