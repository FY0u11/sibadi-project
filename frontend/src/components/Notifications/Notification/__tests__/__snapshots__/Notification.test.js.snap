// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Notification component It should be properly rendered 1`] = `
<DocumentFragment>
  <div
    class="card my-notification"
  >
    <h5
      class="card-header bg-success text-light"
    >
      Test notification
      <button
        class="ml-2 mb-1 close"
        type="button"
      >
        Ã—
      </button>
    </h5>
    <div
      class="card-body"
    >
      <p
        class="card-text"
      >
        Test message
      </p>
      <button
        class="btn btn-success text-light"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
